File: subprojects/docs/src/samples/userguide/plugins/namedDomainObjectContainer/java/buildSrc/src/main/java/DownloadExtension.java
import org.gradle.api.NamedDomainObjectContainer;
import org.gradle.api.model.ObjectFactory;

import javax.inject.Inject;

public class DownloadExtension {
    // A container of `Server` objects
    private final NamedDomainObjectContainer<Server> servers;

    @Inject
    public DownloadExtension(ObjectFactory objectFactory) {
        // Use an injected ObjectFactory to create a container
        servers = objectFactory.domainObjectContainer(Server.class);
    }

    public NamedDomainObjectContainer<Server> getServers() {
        return servers;
    }
}

File: subprojects/docs/src/samples/userguide/plugins/namedDomainObjectContainer/kotlin/buildSrc/src/main/kotlin/DownloadExtension.kt

import javax.inject.Inject
import org.gradle.api.model.ObjectFactory
import org.gradle.api.NamedDomainObjectContainer

open class DownloadExtension @Inject constructor(objectFactory: ObjectFactory) {
    // Use an injected ObjectFactory to create a container of `Server` objects
    val servers = objectFactory.domainObjectContainer(Server::class.java)
}
